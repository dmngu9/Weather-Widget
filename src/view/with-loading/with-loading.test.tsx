import * as React from 'react';
import { shallow } from 'enzyme';

import withLoading from './with-loading';

const Example: React.StatelessComponent<{ title: string }> = ({ title }) => <div>{title}</div>;

const WithLoading = withLoading(Example);

describe('withLoading', () => {
    it('should render correctly when loading', () => {
        const wrapper = shallow(<WithLoading loading title="this is title" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly when not loading', () => {
        const wrapper = shallow(<WithLoading title="this is title" />);

        expect(wrapper).toMatchSnapshot();
    });
});
