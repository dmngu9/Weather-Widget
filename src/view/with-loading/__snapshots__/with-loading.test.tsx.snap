// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoading should render correctly when loading 1`] = `
<Spinner
  delay={100}
  invertColor={false}
  isCompleting={false}
  onComplete={[Function]}
  size="large"
/>
`;

exports[`withLoading should render correctly when not loading 1`] = `
<Example
  title="this is title"
/>
`;
